### PR TITLE
Em 2421 unsaved warning disappear on new multiples

### PR DESCRIPTION
**Pull request by tomxelliott**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: projects/media-viewer/src/lib/annotations/comment-set/comment/comment.component.ts [hunk 1]
```diff
@@ -113,6 +120,7 @@ export class CommentComponent implements OnChanges {
     this._comment.content = this.fullComment.substring(0, this.COMMENT_CHAR_LIMIT);
     this.updated.emit(this._comment);
     this.editable = false;
+    this.hasUnsavedChanges = false;
```
Comment (benouaer) [inline projects/media-viewer/src/lib/annotations/comment-set/comment/comment.component.ts]: this seems to be done twice

File: projects/media-viewer/src/lib/annotations/comment-set/comment/comment.service.ts [hunk 2]
```diff
@@ -20,15 +22,34 @@ export class CommentService {
...
+    if (annotation.comments.length > 0) {
+      const comment = this.getComment(annotation);
+      return comment.hasUnsavedChanges;
+    }
+    return false;
+  }
+
+  updateUnsavedCommentsStatus(annotation: Annotation, editable: boolean): void {
+    const comment = this.getComment(annotation);
+    comment.hasUnsavedChanges = editable;
```
Comment (benouaer) [inline projects/media-viewer/src/lib/annotations/comment-set/comment/comment.service.ts]: why is this the case?

File: projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts [hunk 3]
```diff
@@ -66,9 +68,10 @@ export class AnnotationSetComponent implements OnInit, OnDestroy {
     this.api
       .postAnnotation(annotation)
       .subscribe(newAnnotation => {
+        const unsavedComment = this.commentService.hasUnsavedComments(annotation);
         const index = this.annotationSet.annotations.findIndex(a => a.id === newAnnotation.id);
 
-        this.annotationSet.annotations[index] = newAnnotation;
+        this.annotationSet.annotations[index] = unsavedComment ? annotation : newAnnotation;
```
Comment (velimd) [inline projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts]: im abit confused here. from an over view, you are making a post request and then you are checking if the comment is saved or not and from there you change the annotation. Should this be here?

File: projects/media-viewer/src/lib/annotations/comment-set/comment/comment.service.ts (hunk 2 above)
Comment (tomxelliott) [reply projects/media-viewer/src/lib/annotations/comment-set/comment/comment.service.ts]: I have renamed the variable

File: projects/media-viewer/src/lib/annotations/comment-set/comment/comment.component.ts (hunk 1 above)
Comment (tomxelliott) [reply projects/media-viewer/src/lib/annotations/comment-set/comment/comment.component.ts]: in two separate methods

File: projects/media-viewer/src/lib/annotations/comment-set/comment/comment.component.spec.ts [hunk 4]
```diff
@@ -91,6 +91,13 @@ describe('CommentComponent', () => {
     expect(component.fullComment).toEqual(mockComment.content);
   });
 
+  it('should get comment', () => {
+    component.comment = {...mockComment};
+
+    expect(component._comment).toEqual({...mockComment});
+    expect(component.comment).toEqual({...mockComment});
```
Comment (benouaer) [inline projects/media-viewer/src/lib/annotations/comment-set/comment/comment.component.spec.ts]: I don't get this, you just set it a couple of lines above, so obviously it's gonna be the same

**Pull request merged**